ext/standard/head.c

```diff
@@ -248,24 +224,61 @@ PHP_FUNCTION(setcookie)
 	if (expires_or_options) {
 		if (Z_TYPE_P(expires_or_options) == IS_ARRAY) {
 			if (UNEXPECTED(ZEND_NUM_ARGS() > 3)) {
-				php_error_docref(NULL, E_WARNING, "Cannot pass arguments after the options array");
-				RETURN_FALSE;
+				zend_argument_count_error("%s(): Expects exactly 3 arguments when argument #3 "
+					"($expires_or_options) is an array", get_active_function_name());
+				RETURN_THROWS();
 			}
 			php_head_parse_cookie_options_array(expires_or_options, &expires, &path, &domain, &secure, &httponly, &samesite);
+			if (path && strpbrk(ZSTR_VAL(path), ",; \t\r\n\013\014") != NULL) { /* man isspace for \013 and \014 */
+				zend_value_error("%s(): Argument #3 ($expires_or_options[\"path\"]) cannot contain "
```

Comment:
cc @kocsismate, but I don't think this format makes sense. `$expires_or_options["path"]` is not the parameter name.

Reply:
FWIW, I don't think we should try too hard to get the argument error formatting everywhere. If it's inconvenient from the implementation POV, don't do it.

Reply:
In a similar situation, we used `unserialize(): "max_depth" option must be of type int, %s given` :)